sycl/include/sycl/ext/intel/esimd/memory.hpp

```diff
@@ -1536,14 +1539,25 @@ block_store(T *ptr, simd<T, N> vals, simd_mask<1> pred,
   static_assert(!PropertyListT::template has_property<cache_hint_L3_key>(),
                 "L3 cache hint is reserved. The old/experimental L3 LSC cache "
                 "hint is cache_level::L2 now.");
+  bool ShouldUseOWordDefaultAlign =
```

Comment:
I believe we should not having this dynamic check/jump and generating 2 different versions of block-load for 1 user's block_load() call.
Let's say that because this variant of the function (accepting a predicate) is called, we use the DG2/PVC restrictions.

Reply:
yeah good feedback, i wasnt sure which direction to go, thanks

Reply:
addressed in latest commit hopefully